x-pack/plugins/security_solution/public/resolver/view/panel.tsx

```diff
@@ -3,191 +3,283 @@
+import { useResolverDispatch } from './use_resolver_dispatch';
+import * as event from '../../../common/endpoint/models/event';
+import { ResolverEvent } from '../../../common/endpoint/types';
+import { SideEffectContext } from './side_effect_context';
+import { ProcessEventListNarrowedByType } from './panels/panel_content_related_list';
+import { EventCountsForProcess } from './panels/panel_content_related_counts';
+import { ProcessDetails } from './panels/panel_content_process_detail';
+import { ProcessListWithCounts } from './panels/panel_content_process_list';
+import { RelatedEventDetail } from './panels/panel_content_related_detail';
+import { CrumbInfo } from './panels/panel_content_utilities';
 
-const HorizontalRule = memo(function HorizontalRule() {
-  return (
-    <EuiHorizontalRule
-      style={{
-        /**
-         * Cannot use `styled` to override this because the specificity of EuiHorizontalRule's
-         * CSS selectors is too high.
-         */
-        marginLeft: `-${euiVars.euiPanelPaddingModifiers.paddingMedium}`,
-        marginRight: `-${euiVars.euiPanelPaddingModifiers.paddingMedium}`,
-        /**
-         * The default width is 100%, but this should be greater.
-         */
-        width: 'auto',
-      }}
-    />
-  );
-});
-
-export const Panel = memo(function Event({ className }: { className?: string }) {
-  interface ProcessTableView {
-    name: string;
-    timestamp?: Date;
-    event: ResolverEvent;
-  }
+/**
+ * The team decided to use this table to determine which breadcrumbs/view to display:
+ *
+ * | Crumb/Table            | &crumbId                   | &crumbEvent              |
+ * | :--------------------- | :------------------------- | :----------------------  |
+ * | all processes/default  | null                       | null                     |
+ * | process detail         | entity_id of process       | null                     |
+ * | relateds count by type | entity_id of process       | 'all'                    |
+ * | relateds list 1 type   | entity_id of process       | valid related event type |
+ * | related event detail   | event_id of related event  | entity_id of process     |
+ *
+ * This component implements the strategy laid out above by determining the "right" view and doing some other housekeeping e.g. effects to keep the UI-selected node in line with what's indicated by the URL parameters.
+ *
+ * @returns {JSX.Element} The "right" table content to show based on the query params as described above
+ */
+const PanelContent = memo(function PanelContent() {
+  const history = useHistory();
+  const urlSearch = history.location.search;
+  const dispatch = useResolverDispatch();
 
-  const { processNodePositions } = useSelector(selectors.processNodePositionsAndEdgeLineSegments);
-  const { timestamp } = useContext(SideEffectContext);
+  const queryParams: CrumbInfo = useMemo(() => {
+    return { crumbId: '', crumbEvent: '', ...querystring.parse(urlSearch.slice(1)) };
```

Comment:
I'm confused about defaulting values to `''`. `queryParams` is created here, with values defaulted to empty string. `crumbId` and `crumbEvent` are (later on in this file) destructured into const bindings. I think it's indirect and confusing to have those consts' values be based on this default.

Reply:
per our convo: empty string probably makes sense here. My feedback: maybe this function is just too long. This default value effects code down on line 159 (also in a memo)